### PR TITLE
MAINT: minor refactoring for LogisticRegression

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
Minor refactoring for Ligistic regression.
Moving all sklearnex/onedal4py backend LogReg imports and primitives under daal4py check.
Adresses https://github.com/intel/scikit-learn-intelex/pull/1832#discussion_r1645404928

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by icfaust — APPROVED** [review]
Can merge with green CIs. Though if its a single method, why not just add it to the LogisticRegression module (why does there need to be a Base class?)

**Comment by samir-nasibli** [comment]
> Can merge with green CIs. Though if its a single method, why not just add it to the LogisticRegression module (why does there need to be a Base class?)

To be honest, it’s difficult for me to answer this question now, perhaps it was assumed that this class should have been or will be reused. @avolkov-intel may probably ask better.

**Review by avolkov-intel — APPROVED** [review]

**Comment by samir-nasibli** [comment]
Last Internal CI run: http://intel-ci.intel.com/ef2e8620-9ab6-f164-928f-a4bf010d0e2e
